state-chain/traits/src/lib.rs

```diff
@@ -1323,6 +1348,29 @@ pub trait SpawnAccount {
 	) -> Result<Self::AccountId, DispatchError>;
 }
 
+/// Used in cf_traits and in cf_funding.
+#[derive(Encode, Decode, PartialEq, Debug, TypeInfo, Clone)]
+pub enum FundingSource {
+	EthTransaction { tx_hash: [u8; 32] },
+	Swap { swap_request_id: SwapRequestId },
+	InitialFunding,
+}
+
+pub trait FundAccount {
+	type AccountId;
+	type Amount;
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn get_bond(_account_id: Self::AccountId) -> Self::Amount;
+
+	fn fund_account(
+		account_id: Self::AccountId,
+		funder: Option<cf_chains::eth::Address>,
+		amount: Self::Amount,
+		source: FundingSource,
```

Comment:
IMO we should merge funder and source by changing the funding source to `FundingSource::EthTransaction { tx_hash, funder }`. 

The funder is only `Some` when the source is an eth transaction.